app/connectors/IntegrationFrameworkConnector.scala

```diff
@@ -43,10 +44,22 @@ class IntegrationFrameworkConnector @Inject()(httpClient: HttpClient,
 
   override protected[connectors] val appConfig: AppConfig = appConf
 
+  private def shouldUse2324(taxYear: Int): Boolean = {
+    taxYear == 2024
+  }
+
   def getAllStateBenefitsData(taxYear: Int, nino: String)
                              (implicit hc: HeaderCarrier): Future[Either[ApiError, Option[AllStateBenefitsData]]] = {
-    val url = baseUrl + s"/income-tax/income/state-benefits/$nino/${toTaxYearParam(taxYear)}"
-    val getRequestResponse = callGetStateBenefits(url)(ifHeaderCarrier(url, getApiVersion))
+    if (shouldUse2324(taxYear)) {
+      getAllStateBenefitsData(baseUrl + s"/income-tax/income/state-benefits/23-24/$nino", getApi2324Version)
+    } else {
+      getAllStateBenefitsData(baseUrl + s"/income-tax/income/state-benefits/$nino/${toTaxYearParam(taxYear)}", getApiVersion)
+    }
+  }
+
+  private def getAllStateBenefitsData(url: String, apiVersion: String)
```

Comment:
There is no need for this method. Just return the URL and pass it further, i.e. inline it.
Another approach would be to rename the method to getAllStateBenefitsDataUrl and use it further.